include/rocksdb/encryption.h

```diff
@@ -58,6 +58,12 @@ class KeyManager {
   virtual Status DeleteFile(const std::string& fname) = 0;
   virtual Status LinkFile(const std::string& src_fname,
                           const std::string& dst_fname) = 0;
+  // Provide additional hint of physical file when the key name doesn't map to
+  // one.
+  virtual Status DeleteFileExt(const std::string& fname,
+                               const std::string& /*physical_fname*/) {
```

Comment:
Could you elaborate? I don't understand how the second parameter helps, as it's unused.

Reply:
This is the blanket implementation. TiKV implements virtual function that uses the second parameter.

When keys are stored using original file path, but physical files are renamed to a temporary path, we need the temporary path to scan file system to locate all the keys that need removing.

Reply:
I see.